### PR TITLE
Bool fixes

### PR DESCRIPTION
**Pull request by WarPie**
This works as it should based on all my tests. Now if any inconsistencies of the booleans (numerical) value should show up that should honestly not matter, as all bool evaluation is made to basically check if "bool <> 0". 

Hopefully this "fix" is satisfying enough.

* Construct_If failed, `DoIt = True;` results in LongBool `-1`.. So I changed it to `DoIt:Boolean = True;`

### CODE REVIEW

**Comment by WarPie** [comment]
Okay, the following commit should do it! =)

File: lpeval_functions.inc [hunk 1]
```diff
@@ -2439,11 +2439,25 @@ procedure lpeVariant_ASGN_UString(const Dest, Left, Right: Pointer); {$IFDEF Lap
 procedure lpeBBool_NOT(const Dest, Left, Right: Pointer); {$IFDEF Lape_CDECL}cdecl;{$ENDIF} begin PByteBool(Dest)^ := PByteBool(Left)^ = ByteBool(False); end;
 procedure lpeWBool_NOT(const Dest, Left, Right: Pointer); {$IFDEF Lape_CDECL}cdecl;{$ENDIF} begin PWordBool(Dest)^ := PWordBool(Left)^ = WordBool(False); end;
 procedure lpeLBool_NOT(const Dest, Left, Right: Pointer); {$IFDEF Lape_CDECL}cdecl;{$ENDIF} begin PLongBool(Dest)^ := PLongBool(Left)^ = LongBool(False); end;
+procedure lpeEBool_NOT(const Dest, Left, Right: Pointer); {$IFDEF Lape_CDECL}cdecl;{$ENDIF} begin if PUInt8(Left)^ = 0 then PUInt8(Dest)^ := 1 else PUInt8(Dest)^ := 0; end;
```
Comment (nielsAD) [inline lpeval_functions.inc]: Is there a reason this one differs from the others?

File: lpeval_functions.inc [hunk 2]
```diff
@@ -2439,11 +2439,25 @@ procedure lpeVariant_ASGN_UString(const Dest, Left, Right: Pointer); {$IFDEF Lap
...
+procedure lpeWBool_ASGN_BBool(const Dest, Left, Right: Pointer); {$IFDEF Lape_CDECL}cdecl;{$ENDIF} begin PWordBool(Dest)^ := PUInt8(Left)^  <> 0; end;
+procedure lpeWBool_ASGN_WBool(const Dest, Left, Right: Pointer); {$IFDEF Lape_CDECL}cdecl;{$ENDIF} begin PWordBool(Dest)^ := PUInt16(Left)^ <> 0; end;
+procedure lpeWBool_ASGN_LBool(const Dest, Left, Right: Pointer); {$IFDEF Lape_CDECL}cdecl;{$ENDIF} begin PWordBool(Dest)^ := PUInt32(Left)^ <> 0; end;
+procedure lpeLBool_ASGN_BBool(const Dest, Left, Right: Pointer); {$IFDEF Lape_CDECL}cdecl;{$ENDIF} begin PLongBool(Dest)^ := PUInt8(Left)^  <> 0; end;
+procedure lpeLBool_ASGN_WBool(const Dest, Left, Right: Pointer); {$IFDEF Lape_CDECL}cdecl;{$ENDIF} begin PLongBool(Dest)^ := PUInt16(Left)^ <> 0; end;
+procedure lpeLBool_ASGN_LBool(const Dest, Left, Right: Pointer); {$IFDEF Lape_CDECL}cdecl;{$ENDIF} begin PLongBool(Dest)^ := PUInt32(Left)^ <> 0; end;
+procedure lpeEBool_ASGN_BBool(const Dest, Left, Right: Pointer); {$IFDEF Lape_CDECL}cdecl;{$ENDIF} begin if PUInt8(Left)^  = 0 then PUInt8(Dest)^ := 0 else PUInt8(Dest)^ := 1; end;
```
Comment (nielsAD) [inline lpeval_functions.inc]: Some question here. Would `PBoolean(Dest)^ := PUInt8(Left)^ <> 0;` not work?

File: lpvartypes_ord.pas [hunk 3]
```diff
@@ -633,7 +633,10 @@ function TLapeType_Enum.NewGlobalVarStr(Str: UnicodeString; AName: lpString = ''
 
 function TLapeType_Enum.EvalAsSubType(Op: EOperator; Right: TLapeType): Boolean;
 const
-  BoolOperators = BinaryOperators + EnumOperators - [op_cmp_Equal, op_cmp_NotEqual];
+  BoolOperators = BinaryOperators + EnumOperators - [op_cmp_GreaterThan, op_cmp_GreaterThanOrEqual,
```
Comment (nielsAD) [inline lpvartypes_ord.pas]: I think this boils down to 
`BoolOperators = EnumOperators - CompareOperators - [op_Assign];`
which is equal to
`[op_Plus, op_Minus]`.

Not sure what is more clear. What do you think?

File: lpeval_functions.inc (hunk 2 above)
Comment (WarPie) [inline lpeval_functions.inc]: I'll make that change. I just got a bit tiered of the issue, so I did what was quickest and I _knew_ would work.

File: lpvartypes_ord.pas (hunk 3 above)
Comment (WarPie) [inline lpvartypes_ord.pas]: Myeah, let's make that change. Shorter is better :P

**Comment by nielsAD** [comment]
Thanks! Much appreciated!

**Pull request merged**